pkg/apis/hive/v1alpha1/syncset_types.go

```diff
@@ -116,12 +114,33 @@ type SyncCondition struct {
 	Message string `json:"message,omitempty"`
 }
 
+// SyncSetObjectStatus describes the status of resources created or patches that have
+// been applied from a SyncSet or SelectorSyncSet.
+type SyncSetObjectStatus struct {
+	// Name is the name of the SyncSet.
+	Name string `json:"name"`
+
+	// SyncResources is the list of SyncStatus for objects that have been synced.
+	// +optional
+	Resources []SyncStatus `json:"resources,omitempty"`
+
+	// SyncPatches is the list of SyncStatus for patches that have been applied.
+	// +optional
+	Patches []SyncStatus `json:"patches,omitempty"`
+
+	// Conditions is the list of conditions indicating apply success or failure as well as
+	Conditions []SyncCondition `json:"conditions"`
```

Comment:
Added `conditions` here as non-optional assuming we always want to indicate overall `ApplySuccess`/failure/`UnknownObject`.

Reply:
I'm not sure a general ApplySuccess/Failure condition is useful. We will report those for the individual resources/patches and I think that's enough. This is mainly for the UnknownObject case. I would still leave it as optional.